modules/assim.sequential/R/Analysis_sda_multiSite.R

```diff
@@ -389,45 +447,58 @@ GEF.MultiSite<-function(setting, Forecast, Observed, H, extraArg,...){
   }
 
 
-  dat <-runMCMC(Cmcmc, niter = nitr.GEF, nburnin=nburnin, thin =nthin, nchains = 1)
+  dat <- runMCMC(Cmcmc, niter = nitr.GEF, nburnin = nburnin, thin = nthin, nchains = 1)
+  
+  #---- Saving the chains
+  save(dat, file=paste0("SDA/Chains_", t, ".RData"))
+  
   ## update parameters
   iX   <- grep("Xall[", colnames(dat), fixed = TRUE)
   mu.a <- colMeans(dat[, iX])
   Pa   <- cov(dat[, iX])
   Pa[is.na(Pa)] <- 0
-  
-
   mq <- dat[,  grep("q", colnames(dat))]  # Omega, Precision
   q.bar <- matrix(apply(mq, 2, mean),
                   length(elements.W.Data),
                   length(elements.W.Data)
                   )  # Mean Omega, Precision
-  
-  col <- matrix(1:length(elements.W.Data) ^ 2,
-                length(elements.W.Data),
-                length(elements.W.Data))
-  
-  WV  <- matrix(0, length(elements.W.Data),
-                length(elements.W.Data))
-  
-  for (i in seq_along(elements.W.Data)) {
-    for (j in seq_along(elements.W.Data)) {
-      WV[i, j] <- wish.df(q.bar, X = mq, i = i, j = j, col = col[i, j])
+
+  # Setting up the prior for the next step from the posterior of this step
+  if (t < nt){
+    if (q.type == 1){ #if it's a gamma case
```

Comment:
Is it possible to switch between types while you are doing a run? or are you assuming always staying in one type? What if a pft goes in and out of the data?

Reply:
Right now it's static it's read in line 83 of the same file, but we can discuss this as well next week.

Reply:
@araiho the structure of the process error should stay static -- if you're iteratively updating Q, you can't arbitrarily change the design mid-way. Also, your question about PFT going in-and-out isn't an issue -- this flag controls the across-site structure of Q, not within site. The current design already handles the within-site variability.